### ISSUE TITLE
Netherite with TypeScript Function fails after first yield (Emulating + Deployed)

### ISSUE THREAD

**Issue by lechnerc77** [microsoft/durabletask-netherite#69]
## Issue 

The execution of a Durable Function in emulation mode seems to fail after the first step (`yield`) is executed. The state remains in the `Running` status

## Description

The sample project is the same setup as the one in <https://github.com/microsoft/durabletask-netherite/tree/main/samples/Hello> but using TypeScript.

The intent is to run the Durable Function locally, emulating the EventHub and the Storage. The corresponding settings are stored in the `local.settings.json`.

```json
 "AzureWebJobsStorage": "UseDevelopmentStorage=true",
 "EventHubsConnection": "MemoryF",
```

The emulation of the storage is done via Azurite. 

The extension bundle mechanism is removed and the settings for the extensions are placed in the `host.json`

```json
"extensions": {
    "durableTask": {
      "hubName": "testnetherite",
      "UseGracefulShutdown": true,
      "storageProvider": {
        "type": "Netherite",
        "StorageConnectionName": "AzureWebJobsStorage",
        "EventHubsConnectionName": "EventHubsConnection"
      }
    }
  } 
```

The startup of the function is executed without any errors. When the Durable orchestrator is triggered the first call of the activity gets executed successfully, however the second call is never executed. It seems that the Orchestrator function is not called again. 

The exactly same setup works for the .NET sample. 

The difference that occurred was that no task hubs get created in the emulated storage blob when in the TypeScript case. But even when starting the function with `--verbose` no error occurs during runtime. 

## Expected behavior

The local emulation of Netherite should work for TypeScript as for .NET Core

## Versions

The following version are used:

- Azure Functions Core Tools: 3.0.3734 (Runtime: 3.1.4.0)
- node: 14.16.1
- TypeScript: 4.4.3
- Azurite: 3.14.2

## Sample code

You find the project to reproduce the error here:  <https://github.com/lechnerc77/durablefunctionsnetherite>

**Comment by lechnerc77** [comment]
## Additional Information

I deployed the same app to Azure. The setup is done via the scripts in the sample repository (except that the runtime is changed to node version 14). The Event Hub contains active entities (clients 0..3, partitions), the storage show the structure described in the documentation.

At least to me it seems that the same error occurs in the deployed scenario namely after the first executed activity the orchestrator is not resuming work again.

Due to application insights a bit more information is available i.e. an exception is raised:

_Exception:_  Microsoft.Azure.EventHubs.MessagingEntityNotFoundException
_Message_: The listener for function 'Functions.DurableFunctionsNetheriteOrchestrator' was unable to start. The messaging entity 'sb://testnetheritecl.servicebus.windows.net/partitions' could not be found. 
Failed Method: DurableTask.Netherite.EventHubs.EventHubsConnections+<GetPartitionInformationAsync>d__29.MoveNext

_Call Stack_:

```
Microsoft.Azure.WebJobs.Host.Listeners.FunctionListenerException:
Microsoft.Azure.EventHubs.MessagingEntityNotFoundException:
   at Microsoft.Azure.EventHubs.Amqp.Management.AmqpServiceClient+<GetRuntimeInformationAsync>d__8.MoveNext (Microsoft.Azure.EventHubs, Version=4.3.2.0, Culture=neutral, PublicKeyToken=7e34167dcc6d6d8c)
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw (System.Private.CoreLib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e)
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess (System.Private.CoreLib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification (System.Private.CoreLib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e)
   at Microsoft.Azure.EventHubs.EventHubClient+<GetRuntimeInformationAsync>d__26.MoveNext (Microsoft.Azure.EventHubs, Version=4.3.2.0, Culture=neutral, PublicKeyToken=7e34167dcc6d6d8c)
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw (System.Private.CoreLib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e)
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess (System.Private.CoreLib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification (System.Private.CoreLib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e)
   at DurableTask.Netherite.EventHubs.EventHubsConnections+<GetPartitionInformationAsync>d__29.MoveNext (DurableTask.Netherite, Version=0.0.0.0, Culture=neutral, PublicKeyToken=ef8c4135b1b4225a: C:\source\durabletask-netherite\src\DurableTask.Netherite\TransportProviders\EventHubs\EventHubsConnections.cs:120)
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw (System.Private.CoreLib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e)
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess (System.Private.CoreLib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification (System.Private.CoreLib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e)
   at DurableTask.Netherite.EventHubs.EventHubsConnections+<GetPartitionInfo>d__31.MoveNext (DurableTask.Netherite, Version=0.0.0.0, Culture=neutral, PublicKeyToken=ef8c4135b1b4225a: C:\source\durabletask-netherite\src\DurableTask.Netherite\TransportProviders\EventHubs\EventHubsConnections.cs:167)
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw (System.Private.CoreLib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e)
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess (System.Private.CoreLib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification (System.Private.CoreLib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e)
   at DurableTask.Netherite.EventHubs.EventHubsTransport+<DurableTask-Netherite-ITaskHub-CreateIfNotExistsAsync>d__28.MoveNext (DurableTask.Netherite, Version=0.0.0.0, Culture=neutral, PublicKeyToken=ef8c4135b1b4225a: C:\source\durabletask-netherite\src\DurableTask.Netherite\TransportProviders\EventHubs\EventHubsTransport.cs:106)
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw (System.Private.CoreLib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e)
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess (System.Private.CoreLib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification (System.Private.CoreLib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e)
   at DurableTask.Netherite.NetheriteOrchestrationService+<CreateAsync>d__56.MoveNext (DurableTask.Netherite, Version=0.0.0.0, Culture=neutral, PublicKeyToken=ef8c4135b1b4225a: C:\source\durabletask-netherite\src\DurableTask.Netherite\OrchestrationService\NetheriteOrchestrationService.cs:244)
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw (System.Private.CoreLib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e)
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess (System.Private.CoreLib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification (System.Private.CoreLib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e)
   at DurableTask.Netherite.NetheriteOrchestrationService+<CreateIfNotExistsAsync>d__57.MoveNext (DurableTask.Netherite, Version=0.0.0.0, Culture=neutral, PublicKeyToken=ef8c4135b1b4225a: C:\source\durabletask-netherite\src\DurableTask.Netherite\OrchestrationService\NetheriteOrchestrationService.cs:252)
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw (System.Private.CoreLib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e)
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess (System.Private.CoreLib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification (System.Private.CoreLib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e)
   at Microsoft.Azure.WebJobs.Extensions.DurableTask.DurableTaskExtension+<StartTaskHubWorkerIfNotStartedAsync>d__87.MoveNext (Microsoft.Azure.WebJobs.Extensions.DurableTask, Version=2.0.0.0, Culture=neutral, PublicKeyToken=014045d636e89289: D:\a\r1\a\azure-functions-durable-extension\src\WebJobs.Extensions.DurableTask\DurableTaskExtension.cs:1214)
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw (System.Private.CoreLib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e)
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess (System.Private.CoreLib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification (System.Private.CoreLib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e)
   at System.Runtime.CompilerServices.TaskAwaiter.GetResult (System.Private.CoreLib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e)
   at Microsoft.Azure.WebJobs.Host.Listeners.FunctionListener+<StartAsync>d__13.MoveNext (Microsoft.Azure.WebJobs.Host, Version=3.0.29.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35: C:\projects\azure-webjobs-sdk-rqm4t\src\Microsoft.Azure.WebJobs.Host\Listeners\FunctionListener.cs:69)

```

**Comment by lechnerc77** [comment]
### Update Python

Same result for a Durable Function implemented in Python. The execution stops/is interrupted after the first yielded activity is executed

**Comment by sebastianburckhardt** [comment]
Thanks @lechnerc77 for the detailed repro! I can confirm this is a bug introduced in #43.

The problem is that when running with out-of-proc languages (e.g. Python or TypeScript), orchestrations that are cached in memory cannot resume but must always replay from scratch. Previously to #43, this was always the case because caching of orchestration cursors was controlled by the extendedSessions parameter which was always off for out-of-proc.

Starting with #43, orchestration cursor caching is on by default, which causes the problem you observed.

### Solution

cursor caching should be disabled by default, when using out-of-proc languages.

### Workaround

Adding a `"CacheOrchestrationCursors": "false"` to the host.json fixes the problem:
```
{
  "version": "2.0",
  "extensions": {
    "durableTask": {
      "hubName": "testnetherite",
      "UseGracefulShutdown": true,
      "storageProvider": {
        "type": "Netherite",
        "StorageConnectionName": "AzureWebJobsStorage",
        "EventHubsConnectionName": "EventHubsConnection",
        "CacheOrchestrationCursors": "false",
      }
    }
  }
}
```

**Comment by lechnerc77** [comment]
Great thanks for the quick support!

**Comment by lechnerc77** [comment]
Out of curiosity: does this have an impact on the overall performance when this caching is switched off

**Comment by sebastianburckhardt** [comment]
It depends on the workload. The only real significance of this type of caching is that it skips the replay of histories. This can be significant for orchestrations that have long histories. But in the common case (orchestrations with short histories, or entities) the effect is minor.

**Comment by sebastianburckhardt** [comment]
Fixed with 0.6.0-alpha
